### PR TITLE
Mention the YAML features not supported by the Yaml component

### LINKED ISSUE #6730: Update Yaml docs to list the features not supported
The Yaml component purpose is to parse the config files in Symfony apps. We have added some additional features to make it more useful, but we'll never make it spec-compliant.

Let's improve the Yaml docs adding this note about the purpose of the component and let's list the main unsupported features (such as https://github.com/symfony/symfony/issues/12095).

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #6730 thanks to [this comment](https://github.com/symfony/symfony-docs/issues/6730#issuecomment-313364654) from @GuilhemN.

### CODE REVIEW

**Review by HeahDude — APPROVED** [review]
Very nice!

**Review by xabbuh — APPROVED** [review]

File: components/yaml/yaml_format.rst [hunk 1]
```diff
@@ -171,8 +166,8 @@ Collections
 -----------
 
 A YAML file is rarely used to describe a simple scalar. Most of the time, it
-describes a collection. A collection can be a sequence or a mapping of
-elements. Both sequences and mappings are converted to PHP arrays.
+describes a collection. YAML collections can be a sequence (scalar arrays in PHP)
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: Not sure what you mean by `scalar arrays`. Perhaps `numerically-indexed arrays` would be better?

File: components/yaml/yaml_format.rst [hunk 2]
```diff
@@ -171,8 +166,8 @@ Collections
...
+or a mapping of elements (key/value arrays in PHP).
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: `associative arrays`?

File: components/yaml/yaml_format.rst [hunk 3]
```diff
@@ -318,4 +313,20 @@ The YAML specification defines some tags to set the type of any data explicitly:
...
+The following YAML features are not supported by the Symfony Yaml component:
+
+* Structures and multi-documents (``---`` and ``...`` markers);
+* Complex mapping keys and complex values starting with ``?``;
+* Tagged values as keys;
+* The following tags and types: `!!set`, `!!omap`, `!!pairs`, `!!set`, `!!seq`,
+  `!!bool`, `!!int`, `!!merge`, `!!null`, `!!timestamp`, `!!value`, `!!yaml`;
+* Global tags (``TAG`` directive; example: ``%TAG ! tag:example.com,2000:app/``)
+  and global references to a tag (example: ``!<tag:example.com,2000:app/foo>``);
+* Not having a key in a mapping (example: ``{foo, bar}``; use
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: What about `Using sequence-like syntax for mapping elements`?

File: components/yaml/yaml_format.rst [hunk 4]
```diff
@@ -318,4 +313,20 @@ The YAML specification defines some tags to set the type of any data explicitly:
...
+-------------------------
+
+The following YAML features are not supported by the Symfony Yaml component:
+
+* Structures and multi-documents (``---`` and ``...`` markers);
+* Complex mapping keys and complex values starting with ``?``;
+* Tagged values as keys;
+* The following tags and types: `!!set`, `!!omap`, `!!pairs`, `!!set`, `!!seq`,
+  `!!bool`, `!!int`, `!!merge`, `!!null`, `!!timestamp`, `!!value`, `!!yaml`;
+* Global tags (``TAG`` directive; example: ``%TAG ! tag:example.com,2000:app/``)
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: I'd remove `globals tags` and just keep `TAG directive`

File: components/yaml/yaml_format.rst [hunk 5]
```diff
@@ -318,4 +313,20 @@ The YAML specification defines some tags to set the type of any data explicitly:
...
+  and global references to a tag (example: ``!<tag:example.com,2000:app/foo>``);
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: `global` doesn't feel like the right term... Maybe something like `uri-named tags`?

File: components/yaml/yaml_format.rst [hunk 6]
```diff
@@ -318,4 +313,20 @@ The YAML specification defines some tags to set the type of any data explicitly:
             Pz7Y6OjuDg4J+fn5OTk6enp
             56enmleECcgggoBADs=
 
-.. _YAML: http://yaml.org/
+Unsupported YAML Features
+-------------------------
+
+The following YAML features are not supported by the Symfony Yaml component:
+
+* Structures and multi-documents (``---`` and ``...`` markers);
```
Comment (GuilhemN) [inline components/yaml/yaml_format.rst]: What do you mean by `structures`?

**Review by GuilhemN — COMMENTED** [review]
Thanks for taking care of this!

**Comment by javiereguiluz** [comment]
I've fixed all the reported issues. @GuilhemN thanks for your review.

File: components/yaml/yaml_format.rst [hunk 7]
```diff
@@ -318,4 +313,20 @@ The YAML specification defines some tags to set the type of any data explicitly:
...
+-------------------------
+
+The following YAML features are not supported by the Symfony Yaml component:
+
+* Multi-documents (``---`` and ``...`` markers);
+* Complex mapping keys and complex values starting with ``?``;
+* Tagged values as keys;
+* The following tags and types: `!!set`, `!!omap`, `!!pairs`, `!!set`, `!!seq`,
+  `!!bool`, `!!int`, `!!merge`, `!!null`, `!!timestamp`, `!!value`, `!!yaml`;
+*  Tags (``TAG`` directive; example: ``%TAG ! tag:example.com,2000:app/``)
```
Comment (xabbuh) [inline components/yaml/yaml_format.rst]: one of the spaces needs to be removed between the asterisk and "Tags"

Comment (javiereguiluz) [reply components/yaml/yaml_format.rst]: Fixed. Thanks!

**Comment by xabbuh** [comment]
Thank you @javiereguiluz.

**Pull request closed without merging**